micro-rdk/src/common/data_store.rs

```diff
@@ -253,7 +253,14 @@ impl DataStore for DefaultDataStore {
         }
         let encode_len = message.encoded_len();
         let total_encode_len = length_delimiter_len(encode_len) + encode_len;
-
+        let buffer_capacity = buffer.capacity();
+        if encode_len > buffer_capacity {
+            return Err(DataStoreError::DataTooLarge(
```

Comment:
would we drop the full SensorData object?

Reply:
I believe so, because `write_message` takes ownership of the SensorData object